src/sparseml/export/export_data.py

```diff
@@ -142,9 +112,105 @@ def export_data_sample(
         tensors=samples,
         export_dir=os.path.join(target_path, names.basename.value),
         name_prefix=names.filename.value,
+        break_batch=break_batch,
     )
     if as_tar:
         folder_path = os.path.join(target_path, names.basename.value)
         with tarfile.open(folder_path + ".tar.gz", "w:gz") as tar:
             tar.add(folder_path, arcname=os.path.basename(folder_path))
         shutil.rmtree(folder_path)
+
+
+def create_data_samples(
+    data_loader: torch.utils.data.DataLoader,
+    model: Optional[torch.nn.Module] = None,
+    num_samples: int = 1,
+) -> Tuple[List[Any], List[Any], List[Any]]:
+    """
+    Fetch a batch of samples from the data loader and return the inputs and outputs
+
+    :param data_loader: The data loader to get a batch of inputs/outputs from.
+    :param model: The model to run the inputs through to get the outputs.
+        If None, the outputs will be an empty list.
+    :param num_samples: The number of samples to generate. Defaults to 1
+    :return: The inputs and outputs as lists of torch tensors
+    """
+    inputs, outputs, labels = [], [], []
+    if model is None:
+        _LOGGER.warning("The model is None. The list of outputs will be empty")
+
+    for batch_num, data in tqdm(enumerate(data_loader)):
+        if batch_num == num_samples:
+            break
+        if isinstance(data, dict):
```

Comment:
as requested before, we can run inference on the transformer data but now without the mandatory export from `src.transformers`.